cmake/FindLIBHWLOC.cmake

```diff
@@ -14,6 +14,23 @@ find_file(LIBHWLOC_HEADER NAMES hwloc.h)
 get_filename_component(LIBHWLOC_INCLUDE_DIR ${LIBHWLOC_HEADER} DIRECTORY)
 set(LIBHWLOC_INCLUDE_DIRS ${LIBHWLOC_INCLUDE_DIR})
 
+if(LIBHWLOC_INCLUDE_DIR)
```

Comment:
Please split this PR and extract all HWLOC stuff into a separate PR.

Reply:
Done, https://github.com/oneapi-src/unified-memory-framework/pull/432.